tests/h/settings_test.py

```diff
@@ -98,3 +98,63 @@ def test_mandrill_settings():
 
 def test_mandrill_settings_unset():
     assert settings.mandrill_settings({}) is None
+
+
+class TestSettingsManager(object):
+    def test_does_not_emit_warnings_when_deprecated_setting_is_not_used(self, log):
+        settings_manager = settings.SettingsManager({}, {})
+        settings_manager.set('foo', 'FOO', deprecated_msg='what to do instead')
+
+        assert not log.warn.called
+
+    def test_emits_warnings_when_deprecated_setting_is_used(self, log):
+        settings_manager = settings.SettingsManager({}, {'FOO': 'bar'})
+        settings_manager.set('foo', 'FOO', deprecated_msg='what to do instead')
+
+        result = settings_manager.settings['foo']
+
+        assert result == 'bar'
+        log.warn.assert_called_once_with('use of foo is '
+                                     'deprecated: what to do instead')
```

Comment:
This looks like two separate tests to me:

1. It still sets the setting in `settings`, even if it's deprecated
2. It warns if its deprecated